App/xcbuddykit/Sources/Generator/BuildPhaseGenerator.swift

```diff
@@ -0,0 +1,86 @@
+        }
+    }
+}
+
+/// Build phase generator interface.
+protocol BuildPhaseGenerating: AnyObject {
+    func generateBuildPhases(targetSpec: Target,
+                             target: PBXTarget,
+                             fileElements: ProjectFileElements,
+                             objects: PBXObjects,
+                             context: GeneratorContexting)
+
+    /// Generates the sources build phase.
+    ///
+    /// - Parameters:
+    ///   - buildPhase: build phase specification.
+    ///   - target: target whose build phase is being generated.
+    ///   - fileElements: file elements instance.
+    ///   - objects: project objects.
+    ///   - context: generation context.
+    func generateSourcesBuildPhase(_ buildPhase: SourcesBuildPhase,
+                                   target: PBXTarget,
+                                   fileElements: ProjectFileElements,
+                                   objects: PBXObjects,
+                                   context: GeneratorContexting)
+}
+
+/// Build phase generator.
+final class BuildPhaseGenerator: BuildPhaseGenerating {
+    func generateBuildPhases(targetSpec: Target,
+                             target: PBXTarget,
+                             fileElements: ProjectFileElements,
+                             objects: PBXObjects,
+                             context: GeneratorContexting) {
+        targetSpec.buildPhases.forEach { buildPhase in
+            if let sourcesBuildPhase = buildPhase as? SourcesBuildPhase {
+                generateSourcesBuildPhase(sourcesBuildPhase,
+                                          target: target,
+                                          fileElements: fileElements,
+                                          objects: objects,
+                                          context: context)
+            }
+        }
+    }
+
+    /// Generates the sources build phase.
+    ///
+    /// - Parameters:
+    ///   - buildPhase: build phase specification.
+    ///   - target: target whose build phase is being generated.
+    ///   - fileElements: file elements instance.
+    ///   - objects: project objects.
+    ///   - context: generation context.
+    func generateSourcesBuildPhase(_ buildPhase: SourcesBuildPhase,
+                                   target: PBXTarget,
+                                   fileElements: ProjectFileElements,
+                                   objects: PBXObjects,
+                                   context: GeneratorContexting) {
+        let sourcesBuildPhase = PBXSourcesBuildPhase()
+        let sourcesBuildPhaseReference = objects.addObject(sourcesBuildPhase)
+        target.buildPhases.append(sourcesBuildPhaseReference)
+        buildPhase.buildFiles.files.forEach { path in
+            guard let fileReference = fileElements.file(path: path) else {
+                context.errorHandler.fatal(error: FatalError.bugSilent(BuildPhaseGenerationError.missingFileReference(path)))
```

Comment:
I've followed a different error handling approach in xcbuddy @Juanpe & @esttorhe. When a bug is found, like in this case, we use the error handler to exit the execution reporting the bug. The error handler will print the right output, and report it to Sentry *(when it's a bug)*. If the error is silent, like in this case, nothing is printed to the user because the user cannot do anything in this case to fix the problem. You can see here the types of errors https://github.com/xcbuddy/xcbuddy/blob/master/App/xcbuddykit/Sources/Errors/FatalError.swift